core/logger/logger.go

```diff
@@ -72,6 +75,12 @@ var _ common.Logger = (Logger)(nil)
 //
 // Node Operator Docs: https://docs.chain.link/docs/configuration-variables/#log_level
 // Deprecated: use [common.Logger] & [common.SugaredLogger]
+
+// OtelCore provides access to OTel logger core management functionality
+type OtelCore interface {
+	SetOtelCore(otelLogger otellog.Logger)
+}
```

Comment:
I think we can avoid this additional method by instead leveraging the func for passing in a core. Then we can just hold a reference to the passed `AtomicCore` (or just the `.Store` method) to call later, without doing any casting.